examples/c/ex_col_store.c

```diff
@@ -0,0 +1,555 @@
+        printf("ID %" PRIu64, recno);
+        printf(": hour %" PRIu16 " temp: %" PRIu8 "\n", hour, temp);
+    }
+
+    return true;
+}
+
+/*! [Obtains the average data across all fields given a specific location] */
+void
+average_data(WT_SESSION *session)
+{
+    WT_CURSOR *loc_cursor;
+    const char *day;
+    const char *country;
+    uint64_t recno;
+    uint16_t hour, pressure, loc_long, loc_lat;
+    uint8_t temp, humidity, wind, feels_like_temp;
+    int ret, num_rec = 5, exact;
+    unsigned int count = 0;
+    unsigned int rec_arr[5] = {0, 0, 0, 0, 0};
+
+    /* Open a cursor to search for the location, currently RUS */
+    error_check(
+      session->open_cursor(session, "index:weathertable:country", NULL, NULL, &loc_cursor));
+    loc_cursor->set_key(loc_cursor, "RUS\0\0");
+    error_check(loc_cursor->search_near(loc_cursor, &exact));
+
+    /* Error handling in the case RUS is not found*/
+    if (exact != 0) {
+        printf("Location not found!");
+        return;
+    }
+
+    /* Populate the array with the totals of each of the columns */
+    while ((ret = loc_cursor->next(loc_cursor)) == 0) {
+        error_check(loc_cursor->get_key(loc_cursor, &recno));
+        error_check(loc_cursor->get_value(loc_cursor, &day, &hour, &temp, &humidity, &pressure,
+          &wind, &feels_like_temp, &loc_lat, &loc_long, &country));
+
+        if (!strcmp("RUS", country)) {
+            error_check(loc_cursor->get_key(loc_cursor, &recno));
+            error_check(loc_cursor->get_value(loc_cursor, &day, &hour, &temp, &humidity, &pressure,
+              &wind, &feels_like_temp, &loc_lat, &loc_long, &country));
+            count++;
+
+            /* Increment the values of the rec_arr with the temp_arr values */
+            rec_arr[0] += temp;
+            rec_arr[1] += humidity;
+            rec_arr[2] += pressure;
+            rec_arr[3] += wind;
+            rec_arr[4] += feels_like_temp;
+        }
+    }
+
+    printf("Number of matching entries: %u \n", count);
+
+    /* Get the average values by dividing with the total number of records */
+    for (int i = 0; i < num_rec; i++) {
+        rec_arr[i] = rec_arr[i] / count;
+    }
+    /* List the average records */
+    printf("Average records for location RUS : \nTemp: %" PRIu8 ", Humidity: %" PRIu8
+           ", Pressure: %" PRIu16 ", Wind: %" PRIu8 ", Feels like: %" PRIu8 "\n",
+      rec_arr[0], rec_arr[1], rec_arr[2], rec_arr[3], rec_arr[4]);
+    scan_end_check(ret == WT_NOTFOUND);
+    error_check(loc_cursor->close(loc_cursor));
+}
+
+int
+main(int argc, char *argv[])
+{
+    WEATHER *w;
+    WT_CONNECTION *conn;
+    WT_SESSION *session;
+    WT_CURSOR *cursor;
+    WEATHER weather_data[N_DATA];
+    uint16_t start, end;
+    int min_temp_result, max_temp_result;
+
+    start = 1000;
+    end = 2000;
+    min_temp_result = 0;
+
+    /* Generating random data to populate the weather table */
+    generate_data(weather_data);
+
+    home = example_setup(argc, argv);
```

Comment:
Please setup home before generating data, and move data generation just before you need the data.